File: backend/src/main/java/com/team05/petmeeting/domain/naming/repository/AnimalNameCandidateRepository.java
package com.team05.petmeeting.domain.naming.repository;

import com.team05.petmeeting.domain.naming.entity.AnimalNameCandidate;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface AnimalNameCandidateRepository extends JpaRepository<AnimalNameCandidate, Long>, NamingRepositoryCustom {
    Optional<AnimalNameCandidate> findByAnimalIdAndProposedName(Long animalId, String proposedName);
}

File: backend/src/main/java/com/team05/petmeeting/domain/naming/repository/AnimalNameCandidateRepository.kt
package com.team05.petmeeting.domain.naming.repository

import com.team05.petmeeting.domain.naming.entity.AnimalNameCandidate
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface AnimalNameCandidateRepository : JpaRepository<AnimalNameCandidate, Long>, NamingRepositoryCustom {
    fun findByAnimalIdAndProposedName(animalId: Long, proposedName: String): AnimalNameCandidate?
}
